src/main/java/uk/gov/register/core/RegisterLinkValue.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.register.core;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+/**
+ * A value representing a link. Its "value" is a string representation. It also knows
+ * its target register, and its primary key within the target register.
+ *
+ * For a regular RegisterLinkValue, the "value" and "linkKey" are the same. For a CurieValue, the "value"
+ * is the Curie as a string, while the "linkKey" is the second half of the Curie (after the colon).
+ */
+public class RegisterLinkValue extends LinkValue implements FieldValue {
+    private final RegisterName targetRegister;
+    private final String value;
+    private final String linkKey;
+
+    public RegisterLinkValue(RegisterName registerName, String value) {
+        this(registerName, value, value);
+    }
+
+    private RegisterLinkValue(RegisterName registerName, String value, String linkKey){
+        this.targetRegister = registerName;
+        this.value = value;
+        this.linkKey = linkKey;
+    }
+
+    @Override
+    @JsonValue
+    public String getValue() {
+        return value;
+    }
+
+    @Override
+    public boolean isLinkToRegister() {
+        return true;
+    }
+
+    public RegisterName getTargetRegister() {
+        return targetRegister;
+    }
+
+    public String getLinkKey() {
+        return linkKey;
+    }
+
+    public static class CurieValue extends RegisterLinkValue {
+        public CurieValue(String curieValue) {
+            super(new RegisterName(curieValue.split(":")[0]), curieValue, curieValue.split(":")[1]);
```

Comment:
very minor but I think you only need to do `curieValue.split(":")` once then you can read the two results separately?

Reply:
This code already existed but looks new as I renamed the class. Totally agree though so will change that.

Reply:
@gidsg just realised why it wasn't done this way - it's a constructor which means that `super()` has to be the first statement. Might have to leave it this way.